config/v1/types_infrastructure.go

```diff
@@ -500,6 +506,43 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+// DNSAndLBConfiguration contains information about API and API-Int Load Balancers
+// and who is responsible for that configuration.
+type DNSAndLBConfiguration struct {
```

Comment:
Check the example in openshift/api/example/v1. There's some CEL validation that will be required for the userConfig option for this union

This should also have a `+union` marker on it

Reply:
This still needs the CEL validation from the example to make sure the members of the union are only set when the discriminator has the correct value